### PR TITLE
feat: Spell 2026-09-24 (SC-1682)

### PR DESCRIPTION
**Pull request by supercontracts**
# 2026-09-24 Spark Spell

## Forum Post
https://forum.skyeco.com/t/september-24-2026-proposed-changes-to-spark-for-upcoming-spell/28237

## Dependencies on Sky Core spell
None

## Notes for Reviewers
None

## Conditions for Spell Deployment
- [ ] All spell-specific tests are passing.
- [ ] All E2E tests are passing.
- [ ] All newly deployed addresses have been reviewed and added to the spark-address-registry in a PR that has been merged.
- [ ] All newly deployed addresses have been added to the Immunefi bug bounty program.
- [ ] There is complete test coverage of all changes in the spell, including any relevant edge cases.
- [ ] At least three approvals have been given from reviewers.
- [ ] CODE_RABBIT.md comment has been posted, AI review performed.
- [ ] Octane review performed.

## Conditions for Spell Handoff to Sky
- [ ] All spells payloads have been deployed and verified on Etherscan.
- [ ] All spell-specific tests are passing (including spell bytecode verification).
- [ ] All E2E tests are passing.
- [ ] Spell caster is passing in CI.
- [ ] At least three approvals have been given from reviewers on deployed spell.
- [ ] Comment from at least three reviewers confirming address of mainnet spell payload.

## Conditions for Spell Merge
- [ ] Spell handoff performed to Sky team on Discord.
- [ ] Spell handoff performed to Sky team on Signal.
- [ ] Spell address confirmed to be correct in sky-ecosystem/spells-mainnet repo open PR through comments by at least three reviewers.
- [ ] Sky deployed spell confirmed through a comment in this PR to contain correct Spark payload address by at least three reviewers.

## Summary by CodeRabbit

- **New Features**
  - Added October 2026 funding transfers to the Spark Foundation and Spark Assets Foundation.
  - Allocated USDS to support SPK buybacks through the ALM operations multisig.
  - Configured the Sentora RLUSD Morpho Vault V2 integration.

- **Changes**
  - Updated USDT interest-rate parameters.
  - Completed SparkLend market deprecations on Gnosis, including collateral, E-Mode, liquidation, borrower, and treasury wind-down updates.
  - Deprecated LBTC collateral on Ethereum.
  - Deactivated selected unused liquidity-layer integrations.

### CODE REVIEW

File: src/test-harness/SparkLiquidityLayerTests.sol [hunk 1]
```diff
@@ -4766,51 +4766,11 @@ abstract contract SparkLiquidityLayerTests is SpellRunner {
...
-            "B2C2_TRANSFER-USDT",
-            "B2C2_TRANSFER-PYUSD",
-            "ANCHORAGE_TRANSFER-USAT",
-            "ANCHORAGE_TRANSFER-USDT"
-        ];
-
-        newIntegrations = new SLLIntegration[](integrations.length - offboardedLabels.length + 1);
-
-        uint256 index;
+        newIntegrations = new SLLIntegration[](integrations.length);
```
[the remark this answers is not in the document]
Comment (supercontracts) [reply src/test-harness/SparkLiquidityLayerTests.sol:4739]: This will be done after the previous spark spell executes

File: src/proposals/20260924/SparkEthereum_20260924.sol [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ *         - Transfer the October 2026 monthly grants to the Spark Foundation and the Spark Assets Foundation.
+ *         - Transfer USDS to the buyback executor to fund SPK buybacks.
+ * Forum:
+ */
+contract SparkEthereum_20260924 is SparkPayloadEthereum {
+
+    uint256 internal constant SPARK_FOUNDATION_GRANT_AMOUNT       = 865_000e18;
+    uint256 internal constant SPARK_ASSET_FOUNDATION_GRANT_AMOUNT = 45_000e18;
+
+    uint256 internal constant USDS_SPK_BUYBACK_AMOUNT = 972_485e18;  // TODO: change this amount acc. to proposal passes/not.
```
[the remark this answers is not in the document]
Comment (supercontracts) [reply src/proposals/20260924/SparkEthereum_20260924.sol:23]: Will be done after proposal outcome

File: src/proposals/20260924/Spell_20260924.t.sol [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+}
+
+contract SparkEthereum_20260924_SpellTests is SpellTests {
+
+    uint256 internal constant SPARK_FOUNDATION_GRANT_AMOUNT       = 865_000e18;
+    uint256 internal constant SPARK_ASSET_FOUNDATION_GRANT_AMOUNT = 45_000e18;
+
+    uint256 internal constant USDS_SPK_BUYBACK_AMOUNT = 972_485e18;  // TODO: change this amount acc. to proposal passes/not.
```
Comment (lucas-manuel) [inline src/proposals/20260924/Spell_20260924.t.sol]: Remove TODO

File: src/proposals/20260924/SparkEthereum_20260924.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply src/proposals/20260924/SparkEthereum_20260924.sol]: Remove TODO

File: src/proposals/20260924/SparkEthereum_20260924.sol [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * @title  September 24, 2026 Spark Ethereum Proposal
+ * @author Phoenix Labs
+ * @notice Spark Treasury:
+ *         - Transfer the October 2026 monthly grants to the Spark Foundation and the Spark Assets Foundation.
+ *         - Transfer USDS to the buyback executor to fund SPK buybacks.
+ * Forum:  https://forum.skyeco.com/t/september-24-2026-proposed-changes-to-spark-for-upcoming-spell/28237
+ */
+contract SparkEthereum_20260924 is SparkPayloadEthereum {
```
Comment (lucas-manuel) [inline src/proposals/20260924/SparkEthereum_20260924.sol:18]: Update spark-address-registry to HEAD

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by certora-spark-review — APPROVED** [review]
Approving the spell deployment at: `0xdE40689816DA168b0A56f8F22CBD7FfCFA403E6B`

Codehash:
```
cast codehash 0xdE40689816DA168b0A56f8F22CBD7FfCFA403E6B --rpc-url $MAINNET_RPC_URL
0xf5148b6a9fccbde7f225f4f098c28a112c065205a6bc321418154d41aa384bda
```

Reviewer checklist: https://gist.github.com/certora-spark-review/9b0e78de81a7fa7934cb903b3d9a682f

**Review by lucas-manuel — APPROVED** [review]
Approving deployed spell at `0xdE40689816DA168b0A56f8F22CBD7FfCFA403E6B`.

Codehash
```
cast codehash 0xdE40689816DA168b0A56f8F22CBD7FfCFA403E6B
0xf5148b6a9fccbde7f225f4f098c28a112c065205a6bc321418154d41aa384bda
```

Good to hand over.

https://gist.github.com/lucas-manuel/f337f369ffac7dc77cf51b10304c6cc1

**Comment by deluca-mike** [comment]
Approving deployed spell at `0xdE40689816DA168b0A56f8F22CBD7FfCFA403E6B`.

Codehash
```
cast codehash 0xdE40689816DA168b0A56f8F22CBD7FfCFA403E6B
0xf5148b6a9fccbde7f225f4f098c28a112c065205a6bc321418154d41aa384bda
```

Good to hand over.

https://gist.github.com/deluca-mike/7570323d7cd0ed2669b42d5a57f80dbb

**Review by deluca-mike — APPROVED** [review]

**Review by TheMj0ln1r — APPROVED** [review]
Approving deployed spell at `0xdE40689816DA168b0A56f8F22CBD7FfCFA403E6B`

Codehash

```bash
cast codehash 0xdE40689816DA168b0A56f8F22CBD7FfCFA403E6B
0xf5148b6a9fccbde7f225f4f098c28a112c065205a6bc321418154d41aa384bda
```

Good to handover.

https://gist.github.com/TheMj0ln1r/516f531e23d0aec648de6d0336b36579
